tests/CollectionDataProviderTest.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+use Nayjest\Grids\CollectionDataProvider;
+use Nayjest\Grids\EloquentDataRow;
+use Nayjest\Grids\Grids;
+use Illuminate\Pagination\Paginator;
+use Illuminate\Database\Eloquent\Collection;
+
+class CollectionDataProviderTest extends TestCase
+{
+
+
+    /**
+     * Creates the fake collectiondataprovider for testing
+     *
+     * @return Nayjest\Grids\CollectionDataProvider
+     */
+    private function getTestCollectionDataProvider()
+    {
+        $models = factory(App\User::class, 3)->make();
+        return new CollectionDataProvider($models);
+    }
+
+    /**
+     * Create a Collection Data Provider
+     *
+     * @return \Nayjest\Grids\CollectionDataProvider
+     */
+    public function test_create_CollectionDataProvider()
+    {
+        $this->assertInstanceOf(CollectionDataProvider::class,$this->getTestCollectionDataProvider());
+    }
+
+
+    /**
+     * we should return a paginator
+     *
+     * @return \Illuminate\Pagination\Paginator
+     */
+    public function test_CollectionDataProvider_has_paginator()
+    {
+      $this->assertInstanceOf(Paginator::class,$this->getTestCollectionDataProvider()->getPaginator());
+    }
+
+    /**
+     * we should return a collection
+     *
+     * @return Illuminate\Database\Eloquent\Collection
+     */
+    public function test_CollectionDataProvider_has_collection()
+    {
+      $this->assertInstanceOf(Collection::class,$this->getTestCollectionDataProvider()->getCollection());
+    }
+
+
+    /**
+     * Test Reset Method doesn't throw an error
+     *
+     * @return \Nayjest\Grids\CollectionDataProvider
+     */
+    public function test_reset()
+    {
+        $this->assertInstanceOf(CollectionDataProvider::class,$this->getTestCollectionDataProvider()->reset());
+    }
+
+    /**
+     * Test getPagination Method doesn't throw an error
+     *
+     * @return Illuminate\Pagination\Paginator
+     */
+    public function test_getPaginationFactory()
+    {
+        $this->assertInstanceOf(Paginator::class,$this->getTestCollectionDataProvider()->getPaginationFactory());
+    }
+
+
+    /**
+     * Test getRow Method doesn't throw an error
+     *
+     * @return \Nayjest\Grids\EloquentDataRow
+     */
+    public function test_getRow()
+    {
+        $this->assertInstanceOf(EloquentDataRow::class,$this->getTestCollectionDataProvider()->getRow());
+    }
+
+
+    /**
+     * Test count Method
+     *
+     * @return Int
+     */
+    public function test_count()
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [The method test_count is not named in camelCase.](https://www.codacy.com/app/mail_2/Grids/file/1994350266/issues/source?bid=1990429&fileBranchId=3120437#l93)